library/system/test/commented_config_file_test.rb

```diff
@@ -0,0 +1,314 @@
+      subject { described_class.new }
+
+      it "Splits a simple line with a comment correctly" do
+        expect(subject.split_off_comment("foo = bar # baz")).to eq ["foo = bar", "# baz"]
+      end
+
+      it "Strips leading and trailing whitespace off the content" do
+        expect(subject.split_off_comment("  foo =  bar   # baz")).to eq ["foo =  bar", "# baz"]
+      end
+
+      it "Leaves whitespace in the comment alone" do
+        expect(subject.split_off_comment("foo = bar #  baz  ")).to eq ["foo = bar", "#  baz  "]
+      end
+
+      it "Handles lines without comments well" do
+        expect(subject.split_off_comment("foo = bar")).to eq ["foo = bar", nil]
+      end
+
+      it "Handles comment lines without content well" do
+        expect(subject.split_off_comment("# foo = bar")).to eq ["", "# foo = bar"]
+        expect(subject.split_off_comment("   # foo = bar")).to eq ["", "# foo = bar"]
+      end
+
+      it "Handles empty lines well" do
+        expect(subject.split_off_comment("")).to eq ["", nil]
+      end
+    end
+  end
+
+  context "High-level parser" do
+    describe "#parse" do
+      context "Demo /etc/fstab with header and footer comments" do
+        before(:all) do
+          @file = described_class.new
+          @file.read(TEST_DATA + "fstab/demo-fstab")
+        end
+        subject { @file }
+
+        it "Has the correct header comments" do
+          header = subject.header_comments
+          expect(header.size).to eq 15
+          expect(header[0]).to match /static file system information/
+          expect(header[-2]).to match /mount point.*type.*dump.*pass/
+          expect(header[-1]).to match /^\s*$/
+        end
+
+        it "Has the correct footer comments" do
+          footer = subject.footer_comments
+          expect(footer.size).to eq 1
+          expect(footer[0]).to match /^\s*$/
+        end
+
+        it "Has the correct number of entries" do
+          expect(subject.size).to eq 9
+        end
+
+        it "The first entry is correct" do
+          entry = subject.first
+          expect(entry.content).to match /by-label\/swap\s+none\s+swap/
+          expect(entry.comment_before.size).to eq 1
+          expect(entry.comment_before.first).to match /Linux disk/
+        end
+
+        it "The root filesystem entry is correct" do
+          entry = subject.entries[2]
+          expect(entry.content).to match /Ubuntu.*ext4.*errors=remount-ro/
+          expect(entry.comment_before?).to eq false
+          expect(entry.line_comment?).to eq false
+        end
+
+        it "The last entry is correct" do
+          entry = subject.last
+          expect(entry.content).to match /fritz.box.fritz.nas.*cifs.*forcegid/
+          expect(entry.comment_before.first).to match /^\s*$/
+        end
+      end
+
+      context "Demo /etc/fstab without header and footer comments" do
+        before(:all) do
+          @file = described_class.new
+          puts("TEST_DATA: #{TEST_DATA}")
```

Comment:
Is this need? This might break the `rspec` output, esp. if somebody uses the JSON or HTML output format. (See `rspec --help` for the output formatter options.)

Or make it optional, e.g. `if ENV["DEBUG"]`.

Reply:
Duh - that puts is a leftover of debugging those path settings. Thanks for noticing!